### DISCUSSION TITLE
java sdk for OpenFGA

### DISCUSSION THREAD

**Discussion by udayakr** [openfga/community#31] [Ideas]
Any plans to have java sdk for OpenFGA for integrations with spring/springboot ?

**Comment by adriantam** [comment]
Hello @udayakr , thanks for your question.  Yes, Java SDK and SpringBoot is part of our roadmap item.  Please see https://github.com/orgs/openfga/projects/1/views/1 for details.

**Reply by rhamzeh** [reply]
Created an issue for this under the SDK Generator for more visibility:

https://github.com/openfga/sdk-generator/issues/70

**Comment by ghstahl** [comment]
We have a java based micro services that needs to call OpenFGA.  Saying that, I have decided to follow what Auth0 does internally and front OpenFGA with another micro-service (written in go) that gates access to Write using an OAuth2 token.  OpenFGA at the moment has an all-or-nothing ApiKey approach to calling OpenFGA.  

So, our JAVA SDK, isn't from OpenFGA folks at all.  It's a grpc service whose proto contracts are used to create clients in all our languages (C#, go, Java, TypeScript, Python).  

The only OpenFGA sdk we need is that golang one.

For now, I don't have any auth checks on READ calls, but I have locked down the Write calls into OpenFGA.

**Comment by kdubb** [comment]
Just as an FYI, there is a Quarkus Java SDK (https://github.com/quarkiverse/quarkus-openfga-client).

It's not generated, instead it uses hand coded classes and is built on the Vert.x web client to ensure it's asynchronous and fits into the Quarkus environment.

**Reply by kdubb** [reply]
One of the things we are looking into is replacing the web client with a client generated from gRPC that targets Quarkus/Vert.x gRPC client.

Although, the biggest issue in writing the classes by hand was the authorization model JSON classes which are quite complicated.

**Comment by aaguiarz** [comment]
I'm closing this Idea since it's already in the roadmap, you can keep discussing it here https://github.com/openfga/roadmap/issues/5
